landa/patches/set_phone_and_email_as_primary.py

```diff
@@ -0,0 +1,46 @@
+import frappe
+
+
+def execute():
+    def value_not_set(dict, field):
+        return len(dict) > 0 and (1 not in [d[field] for d in dict])
+
+    contacts = frappe.get_list("Contact", as_list=True)
+    for contact in contacts:
+        # if no email_id is set as primary for a contact, set the first email_id as primary
+        contact_emails = frappe.get_list(
+            "Contact Email",
+            filters={"parent": contact[0]},
+            fields=["name", "is_primary"],
+        )
+        if value_not_set(contact_emails, "is_primary"):
+            frappe.db.set_value(
+                "Contact Email", contact_emails[0]["name"], "is_primary", 1
+            )
+        # if no phone is set as primary mobile or primary phone for a contact,
+        # set all phones as primary phone or primary phone according to the first digits of the phone number
+        contact_phones = frappe.get_list(
+            "Contact Phone",
+            filters={"parent": contact[0]},
```

Comment:
```suggestion
            filters={"parent": contact},
```
Part of the suggestion above.